modules/cli/src/main/java/org/apache/ignite/cli/core/repl/completer/HoconDynamicCompleter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.cli.core.repl.completer;
+
+import com.typesafe.config.Config;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Completes typed words with hocon schema keys.
+ */
+public class HoconDynamicCompleter implements DynamicCompleter {
+
+    private final Config config;
+
+    /** Stores all keys that exist in the given config. Strings are stored in format: "root.subkey.subsubkey". */
+    private final Set<String> leafs;
+
+    /** List of all possible completions. */
+    private final List<String> completions;
+
+    /** Words, after those the completer should have been activated. */
+    private final Set<String> activationPostfixes;
```

Comment:
Should this be called activationPrefixes as in DynamicCompleterFactory?

Reply:
Well, maybe there is a better naming but they are different.

Let's look at the example
`cluster config show --cluster-urt=bleble --some-option --selector=rocksDb,`

`cluster config show` -- prefix

`--selector` --postfix